components/remerge/src/schema/json.rs

```diff
@@ -0,0 +1,1178 @@
+            RawFieldType::Number { .. } => FieldKind::Number,
+            RawFieldType::Integer { .. } => FieldKind::Integer,
+            RawFieldType::Timestamp { .. } => FieldKind::Timestamp,
+            RawFieldType::Boolean { .. } => FieldKind::Boolean,
+            RawFieldType::OwnGuid { .. } => FieldKind::OwnGuid,
+            RawFieldType::UntypedMap { .. } => FieldKind::UntypedMap,
+            RawFieldType::RecordSet { .. } => FieldKind::RecordSet,
+        }
+    }
+
+    pub fn get_merge(&self) -> Option<ParsedMerge> {
+        self.merge().or_else(|| match self {
+            RawFieldType::Timestamp {
+                semantic: Some(RawTimestampSemantic::CreatedAt),
+                ..
+            } => Some(ParsedMerge::TakeMin),
+            RawFieldType::Timestamp {
+                semantic: Some(RawTimestampSemantic::UpdatedAt),
+                ..
+            } => Some(ParsedMerge::TakeMax),
+            _ => None,
+        })
+    }
+
+    pub fn has_default(&self) -> bool {
+        match self {
+            RawFieldType::Untyped { common, .. } => common.default.is_some(),
+            RawFieldType::Text { common, .. } => common.default.is_some(),
+            RawFieldType::Url { common, .. } => common.default.is_some(),
+            RawFieldType::Boolean { common, .. } => common.default.is_some(),
+            RawFieldType::Number { common, .. } => common.default.is_some(),
+            RawFieldType::Integer { common, .. } => common.default.is_some(),
+            RawFieldType::Timestamp { common, .. } => common.default.is_some(),
+            RawFieldType::OwnGuid { .. } => false,
+            RawFieldType::RecordSet { common, .. } => common.default.is_some(),
+            RawFieldType::UntypedMap { common, .. } => common.default.is_some(),
+        }
+    }
+}
+
+/// This (and RawSpecialTime) are basically the same as TimestampDefault, just done
+/// in such a way to make serde deserialize things the way we want for us.
+#[derive(Clone, Copy, Debug, PartialEq, Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum RawTimeDefault {
+    Num(i64),
+    Special(RawSpecialTime),
+}
+
+#[derive(Clone, Copy, Debug, PartialEq, Serialize, Deserialize)]
+pub enum RawSpecialTime {
+    #[serde(rename = "now")]
+    Now,
+}
+#[derive(Clone, Copy, Debug, PartialEq, Serialize, Deserialize)]
+pub enum RawTimestampSemantic {
+    #[serde(rename = "created_at")]
+    CreatedAt,
+
+    #[serde(rename = "updated_at")]
+    UpdatedAt,
+
+    #[serde(other)]
+    Unknown,
+}
+
+impl RawTimestampSemantic {
+    pub fn into_semantic(self) -> Option<TimestampSemantic> {
+        match self {
+            RawTimestampSemantic::CreatedAt => Some(TimestampSemantic::CreatedAt),
+            RawTimestampSemantic::UpdatedAt => Some(TimestampSemantic::UpdatedAt),
+            RawTimestampSemantic::Unknown => None,
+        }
+    }
+}
+struct SchemaParser<'a> {
+    input: &'a RawSchema,
+    #[allow(dead_code)] // XXX USE ME
+    is_remote: bool,
```

Comment:
OOC, where else will this be used, beyond the schema version check in `parse_schema`?

Reply:
I've taken it out for now, but eventually I imagine we might want less strict checks for remote schemas.